### PR TITLE
BHBC-1547: Create Project Improvements

### PR DESCRIPTION
**Pull request by KjartanE**
# Overview

## Links to Jira tickets

- https://quartech.atlassian.net/browse/BHBC-1547

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: app/src/components/formik/ScrollToFormikError.tsx [hunk 1]
```diff
@@ -88,6 +77,11 @@ export const ScrollToFormikError: React.FC<IScrollToFormikErrorProps> = (props)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [errors, submitCount]);
 
+  const closeSnackBar = () =>
+    setOpenSnackbar((currentState) => {
```
Comment (curtisupshall) [inline app/src/components/formik/ScrollToFormikError.tsx:81]: Can the `useEffect` hook in a _functional_ React component legally accept a callback? I tried Googling this and consensus on Stack Overflow seems to be to implement `useEffect`.

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 2]
```diff
@@ -148,7 +135,7 @@ const CreateSurveyPage = () => {
         DATE_FORMAT.ShortDateFormat,
         `Survey start date cannot be before ${getFormattedDate(DATE_FORMAT.ShortMediumDateFormat, DATE_LIMIT.min)}`
       )
-      .required('Required'),
+      .required('Start Data is Required'),
```
Comment (curtisupshall) [inline app/src/features/surveys/CreateSurveyPage.tsx]: Typo - should read "Start Date", not "Start Data".

**Review by curtisupshall — CHANGES_REQUESTED** [review]
Impressive work - code quality is top-notch! 🙌 Just the aforementioned typo change is needed.

**Review by curtisupshall — DISMISSED** [review]
Nicely done 😎

File: app/src/features/projects/create/CreateProjectPrototype.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+/**
+ * Page for creating a new project.
+ *
+ * @return {*}
+ */
+const CreateProjectPrototype: React.FC = () => {
```
Comment (NickPhura) [inline app/src/features/projects/create/CreateProjectPrototype.tsx]: This file can be removed I assume?

File: app/src/features/permits/CreatePermitPage.tsx [hunk 4]
```diff
@@ -65,20 +65,24 @@ export interface IPermitsArrayItem {
...
-  email_address: '',
-  coordinator_agency: '',
-  share_contact_details: 'false',
+  coordinator: {
+    first_name: '',
+    last_name: '',
+    email_address: '',
+    coordinator_agency: '',
+    share_contact_details: 'false'
+  },
```
Comment (NickPhura) [inline app/src/features/permits/CreatePermitPage.tsx:85]: I don't think these changes are used anywhere? Can we revert them, only to reduce conflicts with the other permit branch, if there will be any

**Review by NickPhura — DISMISSED** [review]

**Review by NickPhura — DISMISSED** [review]

File: app/src/features/permits/CreatePermitPage.tsx (hunk 4 above)
Comment (NickPhura) [reply app/src/features/permits/CreatePermitPage.tsx:85]: @KjartanE I ended up reverting this change. I was thinking this was the project permit page, not the stand alone permit page.  Tests broke when these got removed, so easier to just put it back for now, and Alfred can rip it out in his ticket which is removing this functionality entirely anyways.

**Review by NickPhura — APPROVED** [review]

**Pull request merged**